ntbs-service/Services/NotificationService.cs

```diff
@@ -265,13 +267,28 @@ private async Task UpdateDatabase(AuditType auditType = AuditType.Edit)
         }
 
         public IQueryable<Notification> FilterById(IQueryable<Notification> IQ, string IdFilter) {
-            return IQ.Where(s => s.NotificationId.Equals(Int32.Parse(IdFilter)) 
+            int parsedIdFilter;
+            int.TryParse(IdFilter, out parsedIdFilter);
+            return IQ.Where(s => s.NotificationId.Equals(parsedIdFilter) 
                     || s.ETSID.Equals(IdFilter) || s.LTBRID.Equals(IdFilter) || s.PatientDetails.NhsNumber.Equals(IdFilter));
         }
 
         public IQueryable<Notification> OrderQueryableByNotificationDate(IQueryable<Notification> query) {
             return query.OrderByDescending(n => n.CreationDate)
                 .OrderByDescending(n => n.SubmissionDate);
         }
+
+        public async Task<IEnumerable<Notification>> GetNotificationsWithDetailsById(IList<int> ids)
+        {
+            return await repository.GetNotificationsWithDetailsById(ids);
+        }
+
+        public async Task<List<int>> GetPaginatedIdsAsync(
```

Comment:
Hmm, I wonder if this is the right place for it - this method actually has nothing to do with notifications..
You mentioned a "SearchSearvice" at some point - do you think it'd be better off there?
Not sure which ticket is implementing it, but a TODO for that ticket to move it there would be handy!

Reply:
Ah yes - sorry meant to put a comment on the review, Farman and I are working to build a searchservice and searchbuilder as part of our next 2 tickets